### ISSUE TITLE
[clang] False positive clang-analyzer-core.NonNullParamChecker with guard variable and two containers

### ISSUE THREAD

**Issue by chrchr-github** [llvm/llvm-project#101635]
~~~c++
#include <vector>

int f(const std::vector<int>& v) {
	const int* p = nullptr;
	bool b = false;
	std::vector<int> w;
	for (const auto& i : v) {
		if (!b) {
			p = &i;
			b = true;
        }
		else
			w.emplace_back(i);
	}
	int x = 0;
	for (auto j : w) {
		x = std::max(*p, j);
	}
	return x;
}
~~~
`p` is assigned before any element is placed into `w`.
https://godbolt.org/z/WxPv13qz9

**Comment by steakhal** [comment]
It's unfortunate - but true - that we don't make the assumption that a default constructed `vector` is empty - hence it should not take the second loop; so I think we can reduce this case into https://godbolt.org/z/19s5f6chG :
```c++
#include <vector>
int f() {
  const int* p = nullptr;
  std::vector<int> w;
  return w.empty() ? 0 : *p; // FP on the dead ternary branch
}
```
